lib/iris/experimental/ugrid/__init__.py

```diff
@@ -2748,6 +2749,241 @@ def face_node(self):
         return self._members["face_node_connectivity"]
 
 
+def mesh_from_coords(coord_1, coord_2):
```

Comment:
I think we might consider presenting this as a class method `Mesh.from_coords`, similar to `Coord.from_coord` or `DimCoord.from_regular`.
Does that seem slightly neater ?